sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -203,7 +202,11 @@ class UnivocityParser(
     }
   }
 
-  private val doParse = if (options.columnPruning && requiredSchema.isEmpty) {
+  /**
+   * Parses a single CSV string and turns it into either one resulting row or no row (if the
+   * the record is malformed).
+   */
+  val parse: String => Option[InternalRow] = if (options.columnPruning && requiredSchema.isEmpty) {
```

Comment:
It can be just normal method:
```scala
  def parse(input: String): Option[InternalRow] = {
    if (options.columnPruning && requiredSchema.isEmpty) {
      // If `columnPruning` enabled and partition attributes scanned only,
      // `schema` gets empty.
      Some(InternalRow.empty)
    } else {
      // parse if the columnPruning is disabled or requiredSchema is nonEmpty
      convert(tokenizer.parseLine(input))
    }
  }
```

Reply:
In the current way, it saves one comparison for each record vs the suggestion needs to compare for every record. It's maybe no big deal but let me just don't change any behaviours for now in this PR.

Reply:
I see, let's keep it as is. I would be nice to add a comment about the reason.